esmvalcore/typing.py

```diff
@@ -2,10 +2,17 @@
 from __future__ import annotations
 
 from numbers import Number
-from typing import Dict, Sequence, Union
+from typing import Dict, Iterable, Sequence, Union
 
 FacetValue = Union[str, Sequence[str], Number]
 """Type describing a single facet."""
 
 Facets = Dict[str, FacetValue]
 """Type describing a collection of facets."""
+
+NetCDFAttrType = Union[str, Number, Iterable]
```

Comment:
```suggestion
NetCDFAttrType = Union[str, Number, Sequence]
```
since the text mentions they can be sequences? Or even more specific
```suggestion
NetCDFAttrType = Union[str, Number, Sequence[str], Sequence[Number]]
```

Reply:
I originally used `Sequence`, but this lead to problems because `numpy` arrays (which we want to support) are not considered Sequences:

```py
>>> isinstance(np.arange(10), Iterable)
True
>>> isinstance(np.arange(10), Sequence)
False
```

Reply:
That is strange, since they do have the required interface. Let's leave it as is then.